examples/running_mean/main.rs

```diff
@@ -0,0 +1,59 @@
+extern crate linfa;
+extern crate ndarray;
+extern crate ndarray_rand;
+extern crate rand;
+#[macro_use]
+extern crate derive_more;
+
+use crate::standard_scaler::{Config, OnlineOptimizer, ScalingError, StandardScaler};
```

Comment:
So if we have several models, this means we would need to use the full paths, e.g.
```rust
use crate::standard_scaler;
use crate::linear_model::logistic_regression;

let mut standard_scaler_optimizer = standard_scaler::OnlineOptimizer::default();
let standard_scaler = standard_scaler_optimizer.fit(&x, &y, standard_scaler::Config::default())?;
let X_tr, y_tr = optimizer.transform(&x, &y);

let mut logregr_optimizer = logistic_regression::OnlineOptimizer::default();
let standard_scaler = logregr_optimizer.fit(&x, &y, logistic_regression::Config::default())?;
```
which might become somewhat difficult to manage?

Also purely from the user experience and readability (I understand this has other advantages) I find the [builder pattern in rustlearn](https://maciejkula.github.io/rustlearn/doc/rustlearn/linear_models/sgdclassifier/index.html#examples) somewhat simpler because one doesn't have to deal with the optimizer.